RN v12.1.0 for Preview release (#1009)
mia-platform/documentation: docs/release-notes/v12.1.0.md

---
id: v12.1.0
title: Version 12.1.0 Release Notes
sidebar_label: v12.1.0
image: "img/release-note-link-preview.png"
---

_January 18th, 2024_

:::info
Mia-Platform Console v12.1.0 is **now in Preview!**

Console SaaS users can try out v12.1.0 latest improvements in Preview! Open a Service Request to ask for the creation of a sandbox Company in case you do not have access to any Company.

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).
:::

## Console

### New Features

#### New Console backoffice configuration using Mia-Platform Microfrontend Composer

Console CMS has been replaced with a new backoffice adopting Mia-Platform Microfrontend Composer. 
At runtime, there will be micro-lc that replaces the previously adopted cms-backend, cms-site and v1-adapter.

> For self-hosted installations, take a look at the information given on the [How to upgrade Console guide]().

Author: ```suggestion
For self-hosted installations, take a look at the information given on the [How to upgrade Console guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades).
```

#### New User events for the Webhook section

User events have been added that allow a Company Owner to configure and trigger specific webhooks when changes to user access management occur.

Listed below are the events that have been added with this release:

* User added to Company
* User Company role edited.
* User removed from Company
* User added to Project
* User removed from Project

> For more information regarding individual events, please [go to the documentation](/development_suite/webhooks-and-events/events#user-added-to-company).

Author: ```suggestion
For more information regarding individual events, please [go to the documentation](/development_suite/webhooks-and-events/events.mdx#user-added-to-company).
```

#### Complete Company access management through `miactl`

The new version of `miactl` allows complete management of all identities access to the Company.
More specifically, starting from today it will be possible to add, modify and remove any type of identity (User, Service Account, Group) simply through command line.

> For more information on the specific commands introduced, please [go to the documentation](/cli/miactl/30_commands).

Author: ```suggestion
For more information on the specific commands introduced, please [go to the documentation](/cli/miactl/30_commands.md).
```

> 
> ### New Features
> 
> #### New Console backoffice configuration using Mia-Platform Microfrontend Composer
> 
> Console CMS has been replaced with a new backoffice adopting [Mia-Platform Microfrontend Composer](/microfrontend-composer/what-is.md). 
> At runtime, there will be micro-lc that replaces the previously adopted cms-backend, cms-site and v1-adapter.
> 
> For self-hosted installations, take a look at the information given on the [How to upgrade Console guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades).
> 
> #### New User events for the Webhook section
> 
> User events have been added that allow a Company Owner to configure and trigger specific webhooks when changes to user access management occur.
> 
> Listed below are the events that have been added with this release:
> 
> * User added to Company
> * User Company role edited.
> * User removed from Company
> * User added to Project
> * User removed from Project
> 
> For more information regarding individual events, please [go to the documentation](/development_suite/webhooks-and-events/events.mdx#user-added-to-company).
> 
> #### Complete Company access management through `miactl`
> 
> The new version of `miactl` allows complete management of all identities access to the Company.
> More specifically, starting from today it will be possible to add, modify and remove any type of identity (User, Service Account, Group) simply through command line.
> 
> For more information on the specific commands introduced, please [go to the documentation](/cli/miactl/30_commands.md).

Reviewer: What do you think about:

```suggestion

### New Console backoffice configuration using Mia-Platform Microfrontend Composer

Console CMS has been replaced with a new backoffice adopting [Mia-Platform Microfrontend Composer](/microfrontend-composer/what-is.md). 
At runtime, there will be micro-lc that replaces the previously adopted cms-backend, cms-site and v1-adapter.

For self-hosted installations, take a look at the information given on the [How to upgrade Console guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades).

### New User events for the Webhook section

User events have been added that allow a Company Owner to configure and trigger specific webhooks when changes to user access management occur.

Listed below are the events that have been added with this release:

* User added to Company
* User Company role edited.
* User removed from Company
* User added to Project
* User removed from Project

For more information regarding individual events, please [go to the documentation](/development_suite/webhooks-and-events/events.mdx#user-added-to-company).

### Complete Company access management through `miactl`

The new version of `miactl` allows complete management of all identities access to the Company.
More specifically, starting from today it will be possible to add, modify and remove any type of identity (User, Service Account, Group) simply through command line.

For more information on the specific commands introduced, please [go to the documentation](/cli/miactl/30_commands.md).
```

A warning will be displayed in the strategies page if a Single View Creator uses a strategy that does not correspond to any of the strategies linked to that Single View.

##### Prevent removal of Single Views and Projections with attached endpoints

It is not possible anymore to delete a Single View or a Projection with attached endpoints without first deleting said endpoints.

## Microfrontend Composer

### Back-Kit Library

The new version `v1.4.12` of the back-kit library is available!

Refer to back-kit [changelog](/microfrontend-composer/back-kit/changelog.md) for finer-grained details of new versions.

#### Improvements

##### Geopoint field coordinates order can be specified

Geopoint fields support `geopointFormat` option is available in data-schema `formOptions`, allowing to configure whether the Form components read them as `[latitude, longitude]` or `[longitude, latitude]`

#### Bug Fix

This version addressed a few bugs, here is a comprehensive list:

* `bk-calendar` allows to open appointment details in "week" and "day" views by clicking on the respective item

## Marketplace

### Marketplace Updates

## How to update your Console

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `vX.X.X`.

Author: ```suggestion
For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v11.1.0-beta.0`.
```

> For self-hosted installations, please head to the [self hosted upgrade guide](/infrastructure/self-hosted/installation-chart/100-how-to-upgrade.md#v12---version-upgrades) or contact your Mia-Platform referent and upgrade to _Console Helm Chart_ `v11.1.0-beta.0`.

Author: Da confermare la versione della chart qui riportata

---
id: v12.1.0
title: Version 12.1.0 Release Notes
sidebar_label: v12.1.0
image: "img/release-note-link-preview.png"
---

_January 18th, 2023_

:::info
Mia-Platform Console v12.1.0 is **now in Preview!**

Console SaaS users can try out v12.1.0 latest improvements in Preview! Open a Service Request to ask for the creation of a sandbox Company in case you do not have access to any Company.

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).
:::

## Console

### New Features

#### New User events for the Webhook section

> User events have been added that will allow a Company Owner to configure and trigger specific webhooks when user access management changes occur.

Reviewer 2: ```suggestion
User events have been added that allow a Company Owner to configure and trigger specific webhooks when changes to user access management occur.
```

Listed below are the events that have been added with this release:

* User added to Company
* User Company role edited.
* User removed from Company
* User added to Project
* User removed from Project

For more information regarding individual events, please [go to the documentation]().

> #### Complete company access management with the CLI (miactl)
> 
> The new version of miactl allows complete management of all User Company access management.
> In particular from today it will be possible to add, modify and remove any type of identity (User, Service Account or Group) simply through command line.
> 
> For more information on individual commands, please [go to the documentation]().

Reviewer 2: ```suggestion
#### Complete Company access management through `miactl`

The new version of `miactl` allows complete management of all identities access to the Company.
More specifically, starting from today it will be possible to add, modify and remove any type of identity (User, Service Account, Group) simply through command line.

For more information on the specific commands introduced, please [go to the documentation]().
```

---
id: v12.1.0
title: Version 12.1.0 Release Notes
sidebar_label: v12.1.0
image: "img/release-note-link-preview.png"
---

> _January 18th, 2023_

Author: ```suggestion
_January 18th, 2024_
```

For self-hosted installations, please read the [following guidelines](#how-to-update-your-console).
:::

## Console

### New Features

#### New User events for the Webhook section

User events have been added that allow a Company Owner to configure and trigger specific webhooks when changes to user access management occur.

Listed below are the events that have been added with this release:

* User added to Company
* User Company role edited.
* User removed from Company
* User added to Project
* User removed from Project

For more information regarding individual events, please [go to the documentation]().

#### Complete Company access management through `miactl`

The new version of `miactl` allows complete management of all identities access to the Company.
More specifically, starting from today it will be possible to add, modify and remove any type of identity (User, Service Account, Group) simply through command line.

For more information on the specific commands introduced, please [go to the documentation]().

### Improvements

#### Improved User session management in Console

> The User session management in Console has been enhanced, improving the error message displayed when the User session expires.

Reviewer 3: ```suggestion
The Console now features a new modal that activates upon session expiry, preventing further interactions and allowing users to seamlessly re-login, ensuring their ongoing work is not lost.
```
